sphinxlint.py

```diff
@@ -119,6 +119,10 @@
 # .. versionchanged:: 3.6
 three_dot_directive_re = re.compile(r"\.\.\. %s::" % all_directives)
 
+# Find directive missing space after double colon, like:
+# .. versionchanged::3.6
+missing_space_directive_re = re.compile(r"(?<!\.)\.\. %s::[^\s]" % simplename)
```

Comment:
```suggestion
missing_space_directive_re = re.compile(r"(?<!\.)\.\. %s::[^`\s]" % simplename)
```
Would this tweak be enough?

```py
>>> simplename = r"(?:(?!_)\w)+(?:[-._+:](?:(?!_)\w)+)*"
>>> missing_space_directive_re = re.compile(r"(?<!\.)\.\. %s::[^`\s]" % simplename)
>>> missing_space_directive_re.match('`.. foo::123`')
>>> missing_space_directive_re.match('``.. foo::123``')
>>> missing_space_directive_re.match('``.. foo::``')
>>> missing_space_directive_re.match(':dir:`.. foo::`')
>>> missing_space_directive_re.match('.. foo::')
>>> missing_space_directive_re.match('.. foo:: ')
>>> missing_space_directive_re.match('| .. foo:: |')
>>> missing_space_directive_re.match('.. foo::`bar`')
>>> missing_space_directive_re.match('.. foo:: valid')
>>> missing_space_directive_re.match('.. foo::invalid')
<re.Match object; span=(0, 9), match='.. foo::i'>
>>> missing_space_directive_re.match('.. foo::3.7 invalid')
<re.Match object; span=(0, 9), match='.. foo::3'>
```

It catches the last two (invalid), and ignores all the other ones.  It will miss something like `` .. foo::`bar` `` (the user might have wanted `` .. foo:: `bar` ``, which should be valid), but this seems quite unlikely.